app/controllers/post_exports_controller.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class PostExportsController < ApplicationController
+  def index
+    filter_params = {
+      timeframe: params[:timeframe],
+      text: params[:text],
+      with_people: params[:with_people],
+      post_type_ids: params[:post_type_id]&.compact_blank,
+      category_ids: params[:category_ids]&.compact_blank,
+      bookmarked: params[:bookmarked] == '1'
+    }
+
+    filtered_posts = current_user.posts
+                                 .includes(:categories)
+                                 .filter_for_export(**filter_params)
+
+    respond_to do |format|
+      format.csv { send_data filtered_posts.to_csv }
+    end
+  end
+
+  def new
+  end
+
+  def all
```

Comment:
I was not initially a big fan of the approach I took here to add a custom controller action. But, I found that without it, I'd be adding a condition to the `index` action and needing to pass in more params and it just felt messy. So I made the custom action and it's all pretty straightforward as to what's happening here.